src/validation/has.toggle.clicked.ts

```diff
@@ -0,0 +1,45 @@
+import { NextFunction, Request, Response } from "express";
+import { FormattedValidationErrors, ValidationError } from "../model/validation.model";
+import { formatValidationErrors } from "./validation";
+
+
+declare global {
+  namespace Express {
+    interface Response {
+      setErrorMessage: (errors: ValidationError[]) => void;
+      getErrorMessage: (currentUrl: string, referrerUrl: string, lang: string) => FormattedValidationErrors | undefined;
+      setUserData: (req: Request) => void;
+      getUserData: (currentUrl: string, referrerUrl: string) => Request
+    }
+  }
+}
+
+let globalError: ValidationError[] | undefined;
+let globalUserData;
```

Comment:
This will be shared between all calls to the node process as it's global. What if there are two users?

Reply:
@AndreKramer7654 Yes you are right, this will inadvertently mutate requests.
Replaced it with req.app.locals instead